src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayerMP.java

```diff
@@ -55,7 +55,7 @@
 @NonnullByDefault
 @Mixin(EntityPlayerMP.class)
 @Implements(@Interface(iface = Player.class, prefix = "playermp$"))
-public abstract class MixinEntityPlayerMP extends EntityPlayer implements CommandSource {
+public abstract class MixinEntityPlayerMP extends SpongeUser implements CommandSource {
 
```

Comment:
Implemented? No, because you've implemented the ```User``` interface in a separate class which doesn't appear anywhere in the hierarchy of the ```Player``` implementation (mixed in to ```EntityPlayerMP```). I'm not 100% sure what you're actually trying to do here to be honest.